File: Backend/models/map.go
package models

import (
	"time"  
	// "errors" 
	// "../config"
	//"github.com/jinzhu/gorm"  
)
 
// Room ...
type Map struct { 
	ID         int        `json:"id" gorm:"primary_key"` 
	CreatedAt *time.Time `json:"createdAt, omitempty"`
	UpdatedAt *time.Time `json:"updatedAt, omitempty"`
	DeletedAt *time.Time `json:"deletedAt, omitempty" sql:"index"`  
	Name string `json:"name, omitempty" gorm:"not null; type:varchar(100)"`  
	Photo string `json:"photo, omitempty" gorm:"not null; type:varchar(200)"`  
	MapGrap string `json:"cover, omitempty" gorm:"not null; type:varchar(200)"`  
	DataPoints string `json:"datapoints, omitempty" sql:"type:text"`  
	Width float64        `json:"width, omitempty" gorm:"not null"`  
	Height float64        `json:"height, omitempty" gorm:"not null"`  
	Miles	   float64        `json:"miles, omitempty" gorm:"not null"`  
	Status     int        `json:"status, omitempty" gorm:"not null"`  

}

// TableName set Room's table name to be `profiles` ``
func (Map) TableName() string {
	return "maps"
}
 

File: AutonomousSport/src/models/Map.js
export default class Map {
  constructor(mapJson) {
    this.id = mapJson?.id || -1;
    this.userId = mapJson?.userId || 0;
    this.createdAt = mapJson?.createdAt || '';
    this.updatedAt = mapJson?.updatedAt || '';
    this.deletedAt = mapJson?.deletedAt || null;
    this.name = mapJson?.name || '';
    this.photo = mapJson?.photo || '';
    this.cover = mapJson?.cover || '';
    this.mapgrap = mapJson?.mapgrap || '';
    this.miles = mapJson?.miles || '';
    this.datapoints = mapJson?.datapoints || {};
    this.width = mapJson?.width || 0;
    this.height = mapJson.height || 0;
    this.status = mapJson?.status || 0;
  }
  toJSON() {
    return {
      id: this.id,
      userId: this.userId,
      createdAt: this.createdAt,
      updatedAt: this.updatedAt,
      deletedAt: this.deletedAt,
      name: this.name,
      cover: this.cover || '',
      photo: this.photo,
      width: this.width,
      datapoints: this.datapoints || {},
      height: this.height,
      mapgrap: this.mapgrap,
      miles: this.miles,
      status: this.status
    };
  }
}
